### ISSUE TITLE
Unknown argument type when calling `.pop` with default on non-total TypedDict

### ISSUE THREAD

**Issue by Holt59** [microsoft/pyright#5981]
**Describe the bug**

When calling `.pop` on a `TypeDict` created with `total=False` on a dict attribute and passing an empty dict as the default value, an unknown argument type error is reported. This does not happen with `.get()` that should have a similar interface.

**Code or Screenshots**

```python
from typing import TypedDict

A = TypedDict(
    "A",
    {
        "b": dict[str, int],
    },
    total=False,
)


u: A = {}

# no error
u.get("b", {})

# error 'Argument type is partially unknown'
u.pop("b", {})

```

**VS Code extension or command-line**

Both:

- VS Code extension, Pylance v2023.9.12.
- pyright version 1.1.327

**Comment by Holt59** [comment]
Maybe related, but passing a typed variable of completely incorrect type to `.pop` does not trigger an error (also occurs with `.get()` though).

**Comment by erictraut** [comment]
I agree that `pop` should be consistent with `get`. Pyright synthesizes these methods for a `TypedDict` based on the types of its declared fields. I'll update the synthesized (overloaded) signatures for `pop` so it's consistent with `get`.

**Comment by erictraut** [comment]
Thanks for the bug report. This will be addressed in the next release.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.328, which I just published. It will also be included in a future release of pylance.
